ivy/functional/ivy/linear_algebra.py

```diff
@@ -340,16 +343,124 @@ def diagonal(
     >>> d = ivy.diagonal(x, 1)
     >>> print(d)
     ivy.array([1.])
+    >>> x = ivy.array([[[1., 2.],\
+                        [3., 4.]],\
+                       [[5., 6.],\
+                        [7., 8.]]])
+    >>> d = ivy.diagonal(x)
+    >>> print(d)
+    ivy.array([[1.,4.],
+               [5.,8.]])
 
-    A 3-D Example
+    Functional examples
```

Comment:
Thanks for making the edits! Just a few more changes to make:

- [ ] 1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example

- [ ] 2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.

- [ ] 3. Insert a line-space between each example.

- [ ] 4. I've noted two lines that you should delete so please delete those.

- [ ] 5.  Include a functional example that takes an `ivy.Container` input

- [ ] 6.  Include an instance method example that takes an `ivy.Container` instance method

Reply:
> Thanks for making the edits! Just a few more changes to make:
> 
> * [ ]  1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example
> * [ ]  2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.
> * [ ]  3. Insert a line-space between each example.
> * [ ]  4. I've noted two lines that you should delete so please delete those.
> * [ ]  5.  Include a functional example that takes an `ivy.Container` input
> * [ ]  6.  Include an instance method example that takes an `ivy.Container` instance method

Hello @chie2727, thanks for the valuable suggestions, I have made the necessary changes please let me know if anything else has to be changed.

Reply:
> Thanks for making the edits! Just a few more changes to make:
> 
> * [ ]  1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example
> * [ ]  2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.
> * [ ]  3. Insert a line-space between each example.
> * [ ]  4. I've noted two lines that you should delete so please delete those.
> * [ ]  5.  Include a functional example that takes an `ivy.Container` input
> * [ ]  6.  Include an instance method example that takes an `ivy.Container` instance method

I am trying to tab on button for re review request but, it is not working, so commenting here. @chie2727